src/utils/dashboardModelUtils.ts

```diff
@@ -45,6 +45,31 @@ export function removeWidgetFromState(contents: DashboardContentsModel, id: stri
   return new_contents;
 }
 
+export function moveWidget(
+  contents: DashboardContentsModel,
+  sourceIndex: number,
+  sourceContainerId: string,
+  destinationIndex: number,
+  destinationContainerId: string,
+  widgetId: string
+) {
+  let rootSection = contents.rootSection;
+  let sections = {...contents.sections};
+  let widgets = {...contents.widgets};
+  let newContents : DashboardContentsModel = {rootSection, sections, widgets};
+
+  newContents = removeChildAndReDistributeSizing(newContents, sourceContainerId, widgetId);
+  newContents = addWidgetToState(newContents, destinationContainerId, widgetId, destinationIndex)
```

Comment:
Omg reusing my removeChildAndRedistributeSizing <3 <3 <3

Reply:
Omg writing great re-usable functions in the first place <3 <3 <3 u made it so easy